lock-impl/src/main/java/com/palantir/lock/logger/LockServiceStateLogger.java

```diff
@@ -0,0 +1,216 @@
+    }
+
+    private List<SimpleLockRequest> getDescriptorSimpleRequestMap(LockClient client, LockRequest request) {
+        return request.getLocks().stream()
+                .map(lock ->
+                        SimpleLockRequest.of(request,
+                                this.lockDescriptorMapper.getDescriptorMapping(lock.getLockDescriptor().getLockIdAsString()),
+                                client.getClientId()))
+                .collect(Collectors.toList());
+    }
+
+    private DumperOptions getDumperOptions() {
+        DumperOptions options = new DumperOptions();
+        options.setDefaultFlowStyle(DumperOptions.FlowStyle.BLOCK);
+        options.setIndent(4);
+        options.setAllowReadOnlyProperties(true);
+        return options;
+    }
+
+    private Representer getRepresenter() {
+        Representer representer = new Representer();
+        representer.addClassTag(ImmutableSimpleTokenInfo.class, Tag.MAP);
+        representer.addClassTag(ImmutableSimpleLockRequest.class, Tag.MAP);
+        representer.addClassTag(SimpleLockRequestsWithSameDescriptor.class, Tag.MAP);
+        return representer;
+    }
+
+    private Map<String, Object> getDescriptorToTokenMap(HeldLocksToken heldLocksToken,
+            LockServiceImpl.HeldLocks<HeldLocksToken> heldLocks) {
+        Map<String, Object> lockToLockInfo = Maps.newHashMap();
+        heldLocks.getLockDescriptors()
+                .forEach(lockDescriptor ->
+                        lockToLockInfo.put(
+                                this.lockDescriptorMapper.getDescriptorMapping(lockDescriptor.getLockIdAsString()),
+                                SimpleTokenInfo.of(heldLocksToken)));
+        return lockToLockInfo;
+    }
+
+    private void dumpYamlsInNewFiles(String outputDir, Map<String, Object> generatedOutstandingRequests,
+            Map<String, Object> generatedHeldLocks) throws IOException {
+        String fileName = "lockstate-" + this.startTimestamp + ".yaml";
+        File file = new File(outputDir, fileName);
+
+        String descriptorsFileName = "descriptors-" + this.startTimestamp + ".yaml";
+        File descriptorsFile = new File(outputDir, descriptorsFileName);
+
+        createNewFile(file);
+        createNewFile(descriptorsFile);
+
+        dumpYaml(generatedOutstandingRequests, generatedHeldLocks, file);
+        dumpDescriptorsYaml(descriptorsFile);
+    }
+
+    private void createNewFile(File file) throws IOException {
+        if (!file.createNewFile()) {
+            log.error(FILE_NOT_CREATED_LOG_ERROR, file.getAbsolutePath());
+            throw new IllegalStateException(YAML_FILE_ALREADY_EXISTS_EXCEPTION + file.getAbsolutePath());
+        }
+    }
+
+    private void dumpYaml(Map<String, Object> generatedOutstandingRequests, Map<String, Object> generatedHeldLocks,
+            File file) throws IOException {
+        FileWriter writer = new FileWriter(file);
+        Yaml yaml = new Yaml(getRepresenter(), getDumperOptions());
```

Comment:
Can we group all our Yaml config into it's own class that we just pass a file and an object to dump.  Will make start to move this class toward only have one responsibility.